optuna/study.py

```diff
@@ -505,7 +516,9 @@ def create_study(
         study_name=study_name,
         storage=storage,
         sampler=sampler,
-        pruner=pruner)
+        pruner=pruner,
+        enable_storage_cache=enable_storage_cache,
```

Comment:
I think we can write the code below that has inconsistent `enable_storage_cache` argument values because both `optuna.storages.get_storage` and `optuna.create_study` have `enable_storage_cache`.

```python
storage = optuna.storages.get_storage('sqlite:///foo.db', enable_storage_cache=False)
study = optuna.create_study(storage=storage, enable_storage_cache=True)
print('cache enabled: ', study.storage._finished_trials_cache._enabled)  # False
```

Do you think it is acceptable?

I don't think ordinal users write such codes, but this may make developers confused.
For example, let's assume that developers create a web-service which executes hyperparameter tuning when it receives a request. In the service, they may want to reuse a storage object, and the reused storage object will cause the inconsistency shown above.

I may be too worried, but please let me know if you have any ideas to solve it.

Reply:
We discussed this issue offline and decided to remove `enable_storage_cache` option except for `RDBStorage` to preserve the consistency.

Reply:
Done: https://github.com/pfnet/optuna/pull/349/commits/05ce9c70c653c6890d22dbf50766e3cb0392a8b0

Besides, in this commit, I changed the option name from `enable_storage_cache` to `enable_cache` (the word "storage" in the name seemed redundant since the option became to be only used in `RDBStorage`).